File: mobile/backend/app/schemas/report.py
"""
Report schemas for request/response validation.
"""

from typing import Optional
from pydantic import BaseModel, Field


class ReportCreate(BaseModel):
    """Schema for creating a report."""
    operation_id: str
    missing_quantity: int = Field(default=0, ge=0)
    physical_damage: bool = False
    extra_quality: int = Field(default=0, ge=0)


class ReportResponse(BaseModel):
    """Schema for report response."""
    id: str
    operation_id: str
    missing_quantity: int
    physical_damage: bool
    extra_quality: int
    created_at: Optional[str] = None
    updated_at: Optional[str] = None

    class Config:
        from_attributes = True

File: mobile/frontend/lib/models/report.dart
class Report {
  final String id;
  final String operationId;
  final bool physicalDamage;
  final int missingQuantity;
  final int extraQuality;
  final String? reportedBy;
  final String? createdAt;
  final String? updatedAt;

  Report({
    required this.id,
    required this.operationId,
    this.physicalDamage = false,
    this.missingQuantity = 0,
    this.extraQuality = 0,
    this.reportedBy,
    this.createdAt,
    this.updatedAt,
  });

  factory Report.fromJson(Map<String, dynamic> json) {
    return Report(
      id: json['id'] ?? '',
      operationId: json['operation_id'] ?? '',
      physicalDamage: json['physical_damage'] ?? false,
      missingQuantity: json['missing_quantity'] ?? 0,
      extraQuality: json['extra_quality'] ?? 0,
      reportedBy: json['reported_by'],
      createdAt: json['created_at'],
      updatedAt: json['updated_at'],
    );
  }

  Map<String, dynamic> toJson() => {
    'operation_id': operationId,
    'physical_damage': physicalDamage,
    'missing_quantity': missingQuantity,
    'extra_quality': extraQuality,
  };
}
